sympy/logic/boolalg.py

```diff
@@ -284,9 +363,71 @@ class BooleanFunction(Application, Boolean):
     It is used as base class for And, Or, Not, etc.
     """
     is_Boolean = True
+    def __lt__(self, other):
+        from sympy.logic.boolalg import Boolean
+        if isinstance(self, Boolean):
```

Comment:
This will always be True. BooleanFunction is a subclass of Boolean.

Reply:
Corrected and tested.